corehq/ex-submodules/pillowtop/processors/form.py

```diff
@@ -104,4 +118,14 @@ def mark_latest_submission(domain, user_id, app_id, build_id, version, metadata,
         last_submission.build_version = app_version_info.build_version
         last_submission.commcare_version = app_version_info.commcare_version
 
+        if app_version_info.build_version:
+            update_latest_builds(user, app_id, received_on_datetime, app_version_info.build_version)
+
+        if _last_submission_needs_update(user.reporting_metadata.last_submission_for_user,
+                                         received_on_datetime,
+                                         app_version_info.build_version,
+                                         app_version_info.commcare_version):
```

Comment:
I think I agree with you but I still think the code is wrong (though I think my original reasoning is also not quite right).

I think the debounce issue is confusing when we're looking at 2 data models (global last submission and per app last submission). I think the reason you're using the per-app submission in the outer `if` is to make sure we get the 'most recent - within 15 min etc' submission for the app. If we'd used the global submission as the debounce model then we could get a submission from one app and then from another and the second would be missed. Correct?

I don't think the inner `if` is necessary at all since we don't care about debounce at that point - we only care if this submission is after the user's global latest.